### PR TITLE
Animate transition for chat message actions

### PR DESCRIPTION
**Pull request by fortmarek**
## Description of the pull request

Add transition animation for `ChatMessagePopupVC`. This also moves the `messageContentView` as a whole, so it should keep e.g. GIF playing (currently, it was being reloaded)

### CODE REVIEW

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageContentView.swift [hunk 1]
```diff
@@ -147,6 +147,9 @@ open class _ChatMessageContentView<ExtraData: ExtraDataTypes>: _View, ThemeProvi
     /// The container which holds `onlyVisibleForYouIconImageView` and `onlyVisibleForYouLabel`
     public private(set) var onlyVisibleForYouContainer: ContainerStackView?
 
+    /// Constraint between bubble and reactions.
+    public private(set) var bubbleToReactionsConstraint: NSLayoutConstraint?
+
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageContentView.swift:152]: What was the reason for this?

File: Sources/StreamChatUI/MessageActionsPopup/ChatMessagePopupVC.swift [hunk 2]
```diff
@@ -62,17 +60,19 @@ open class _ChatMessagePopupVC<ExtraData: ExtraDataTypes>: _ViewController, Comp
     }
 
     override open func setUpLayout() {
-        scrollView.embed(scrollContentView)
+        guard messageViewFrame != nil else { return }
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/MessageActionsPopup/ChatMessagePopupVC.swift:63]: We shouldn't silently fail here but log an error with a description of what's happening instead.

File: Sources/StreamChatUI/MessageActionsPopup/ChatMessagePopupVC_Tests.swift [hunk 3]
```diff
@@ -49,7 +49,16 @@ final class ChatMessagePopupVC_Tests: XCTestCase {
...
+        vc.messageViewFrame = CGRect(x: 50, y: 300, width: 200, height: 200)
+        vc.messageViewFrame.size = vc.messageContentView.systemLayoutSizeFitting(
+            CGSize(width: vc.messageViewFrame.width, height: UIView.layoutFittingCompressedSize.height),
+            withHorizontalFittingPriority: .streamRequire,
+            verticalFittingPriority: .streamLow
+        )
+        
+        vc.messageContentContainerView.addSubview(messageContentView)
+        messageContentView.translatesAutoresizingMaskIntoConstraints = false
+        vc.messageContentContainerView.embed(messageContentView)
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/MessageActionsPopup/ChatMessagePopupVC_Tests.swift:61]: Either `addSubview` or `embed` is redundant here

File: Sources/StreamChatUI/MessageActionsPopup/ChatMessagePopupVC_Tests.swift [hunk 4]
```diff
@@ -114,6 +127,14 @@ final class ChatMessagePopupVC_Tests: XCTestCase {
...
         vc.messageContentView = messageContentView
         vc.messageViewFrame = CGRect(x: 50, y: 300, width: 200, height: 50)
+        vc.messageViewFrame.size = vc.messageContentView.systemLayoutSizeFitting(
+            CGSize(width: vc.messageViewFrame.width, height: UIView.layoutFittingCompressedSize.height),
+            withHorizontalFittingPriority: .streamRequire,
+            verticalFittingPriority: .streamLow
+        )
+        vc.messageContentContainerView.addSubview(messageContentView)
+        messageContentView.translatesAutoresizingMaskIntoConstraints = false
+        vc.messageContentContainerView.embed(messageContentView)
```
Comment (VojtaStavik) [inline Sources/StreamChatUI/MessageActionsPopup/ChatMessagePopupVC_Tests.swift:137]: same here

**Review by VojtaStavik — APPROVED** [review]
LGTM, thanks! Can you please double check the snapshots are correct? They don't look correct to me :) 

Screen Shot 2021-05-07 at 13 26 12

File: Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageContentView.swift (hunk 1 above)
Comment (fortmarek) [reply Sources/StreamChatUI/ChatMessageList/ChatMessage/ChatMessageContentView.swift:152]: I needed to turn off the constraint to layout the message content view _without_ reactions. Let me know if there is more elegant way how to achieve this. I have pondered using `func layout(options: ChatMessageLayoutOptions)` but AFAICT this method can be called only once?

**Pull request merged**